### ISSUE TITLE
Provide documentation example showing ServiceInstance with pre-defined UID

### ISSUE THREAD

**Issue by ivandov** [kubernetes-sigs/service-catalog#2502]
In certain scenarios, a user may want to force the usage of a static UID for a new ServiceInstance. For example, when developing a broker, and requiring a static instance ID that may be recreated in dev/test scenarios. 

This is possible via manually invoking the OSB APIs directly, but it would be very beneficial if this can be done directly in a YAML definition as well. Potentially some flag for a forced UID. I tried to set the `metadata.UID` and `spec.external id` fields directly, but those are overwritten in the creation process.

**Comment by ivandov** [comment]
I found a working example from this PR to the walkthrough docs: 

https://github.com/kubernetes-incubator/service-catalog/pull/839/files?short_path=2af5bed#diff-2af5bedd4fd75eebf48a33be7b0a3eb8

apparently the proper fields to set are `metadata.uid` and/or `spec.externalID`, I set both to the same value, though, I'm not sure which one actually caused the static ID to be set as I had expected. 

Again, this would be valuable documentation in the walkthrough, again, for more advanced scenarios. Typical users of a registered broker should not care / be-aware-of a `ServiceInstance's` UID.
